### ISSUE TITLE
Sound issues for vamphalf driver

### ISSUE THREAD

**Issue by Panderner** [finalburnneo/FBNeo#322]
Vamphalf is recently added on FBNeo. But it lacks OKI MSM6295 for most games and crackly sound in mission craft.

**Comment by dinkc64** [comment]
Hi,
I improved the sound for some games in vamphalf, please give it a try and if you notice something wrong let me know exactly which game(s) have problems.
best regards,
- dink

**Comment by Panderner** [comment]
But there's is no OKI MSM6295 I heard for most games in latest version

**Comment by Panderner** [comment]
And the OKI MSM6295 is unimplemented in vamphalf driver

**Comment by barbudreadmon** [comment]
> Because the OKI MSM6295 is unimplemented in vamphalf driver

No, it is implemented

**Comment by barbudreadmon** [comment]
@Panderner it appears the sound isn't working properly if samplerate is set to 48000 (which is the default in the libretro port), for now you can set it to 44100 as a workaround.

**Comment by Panderner** [comment]
OKI MSM6295 Doesn't work when the sample rate set to 48000hz

**Comment by barbudreadmon** [comment]
> I set to 44100hz but it does not help mission craft

The crackling ? It was fixed in https://github.com/finalburnneo/FBNeo/commit/1a06c6ffc97d2e5a4fb7bdb3e4ec41e564e8255e , but it's probably not on the libretro buildbot yet.

**Comment by dinkc64** [comment]
Both sound bugs should be fixed now, it's just a matter of time for them to make it through the build-bot.

best regards,
- dink

**Comment by Panderner** [comment]
Mission craft sounds fixed but it's too loud

**Comment by dinkc64** [comment]
good to hear! I will adjust the volume, let me know if it helps.

**Comment by Panderner** [comment]
But the problem is the wivern wings music, the drums is too quiet.

**Comment by dinkc64** [comment]
We can have a choice: low drums in Wivern Wings with normal/good sounding SFX in Wivern Wings and Mission Craft, or loud drums in Wivern Wings but low/bad sounding SFX in both games.
I can try some more to improve the sound, but its very difficult - this sounddevice is very incomplete, and there are no documentation anywhere for it (AdMOS QS1000) :(

**Comment by Panderner** [comment]
QS1000 is a General MIDI Sound Card. And it is used for some drivers like wivern wings, mission craft, limenko and eolith's own arcade hardware.

**Comment by Panderner** [comment]
And here's a gameplay of wivern wings running on MAME: https://youtu.be/Y-GcmydGCj8

**Comment by dinkc64** [comment]
Clarification: QS1000 is a sample processor chip, if it is used on a soundcard with midi, the midi part of course comes from a midi processor chip.  

Sure, I'll make it sound like that video for you.

**Comment by dinkc64** [comment]
Hows it now? :)

**Comment by Panderner** [comment]
The sounds gets better I'm closing now

**Comment by litaniast** [comment]
https://www.youtube.com/watch?v=zyiikNKtnuk

Arcade PCB video of Mission Craft. It isn't too loud music volume (unlike MAME/FBN)

**Comment by Panderner** [comment]
I can't find actually arcade PCB capture of wyvern wings that isn't on MAME.

**Comment by barbudreadmon** [comment]
Anyway, it can't be improved with the material we got.

**Comment by dinkc64** [comment]
I've heard wivern wings (orig pcb), it sounds really beautiful.  It will be impossible to achieve the level of sound without more information on the internal workings of the QS1000 chip, though.  I spent about 3 solid days trying to improve it - only to hit a dead end.....
While I found the electronic datasheet for the QS1000 chip... it tells nothing of how the chip works internally - especially the registers, modulation (filters, env, etc), how are the samples data arranged - etc.
